x-pack/plugins/data_visualizer/public/application/data_comparison/document_count_with_dual_brush.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { WindowParameters } from '@kbn/aiops-utils';
+import React, { FC } from 'react';
+import { DocumentCountChart, type DocumentCountChartPoint } from '@kbn/aiops-components';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import type { DocumentCountChartProps } from '@kbn/aiops-components';
+import { RandomSampler } from '@kbn/ml-random-sampler-utils';
+import { useDataVisualizerKibana } from '../kibana_context';
+import { DocumentCountStats } from '../../../common/types/field_stats';
+import { TotalCountHeader } from '../common/components/document_count_content/total_count_header';
+import { SamplingMenu } from '../common/components/random_sampling_menu/random_sampling_menu';
+
+export interface DocumentCountContentProps
+  extends Omit<
+    DocumentCountChartProps,
+    | 'dependencies'
+    | 'chartPoints'
+    | 'timeRangeEarliest'
+    | 'timeRangeLatest'
+    | 'interval'
+    | 'chartPointsSplitLabel'
+  > {
+  brushSelectionUpdateHandler: (d: WindowParameters, force: boolean) => void;
+  documentCountStats?: DocumentCountStats;
+  documentCountStatsSplit?: DocumentCountStats;
+  documentCountStatsSplitLabel?: string;
+  isBrushCleared: boolean;
+  totalCount: number;
+  sampleProbability: number;
+  initialAnalysisStart?: number | WindowParameters;
+  /** Optional color override for the default bar color for charts */
+  barColorOverride?: string;
+  /** Optional color override for the highlighted bar color for charts */
+  barHighlightColorOverride?: string;
+  windowParameters?: WindowParameters;
+  incomingInitialAnalysisStart?: number | WindowParameters;
+  randomSampler: RandomSampler;
```

Comment:
Should the document count chart re-render when the random sampling configuration is changed (as it does in the data visualizer). Currently it doesn't re-query/render so you can end up in this slightly confusing state where the count is in an inconsistent state with the control:

<img width="1369" alt="image" src="https://github.com/elastic/kibana/assets/7405507/392248c5-6efe-45bb-a277-eecbecf9fc56">

Reply:
Fixed here [`509a389` (#161365)](https://github.com/elastic/kibana/pull/161365/commits/509a3898398d226db1634dd468bde885f9007823) 

https://github.com/elastic/kibana/assets/43350163/e4af7d33-fc35-475f-ba17-41f7ee233a23